DatadogLogs/Sources/Logger.swift

```diff
@@ -7,247 +7,183 @@
-    ///   - error: the error information (optional)
-    ///   - attributes: a dictionary of attributes (optional) to add for this message. If an attribute with
-    /// the same key already exist in this logger, it will be overridden (only for this message).
-    func notice(_ message: String, error: Error? = nil, attributes: [AttributeKey: AttributeValue]? = nil) {
-        log(level: .notice, message: message, error: error, attributes: attributes)
-    }
+/// Datadog logger.
+public struct Logger {
+    public struct Configuration {
+        /// Format to use when printing logs to console.
+        public enum ConsoleLogFormat {
+            /// Prints short representation of log.
+            case short
+            /// Prints short representation of log with given prefix.
+            case shortWith(prefix: String)
+        }
 
-    /// Sends a WARN log message.
-    /// - Parameters:
-    ///   - message: the message to be logged
-    ///   - error: the error information (optional)
-    ///   - attributes: a dictionary of attributes (optional) to add for this message. If an attribute with
-    /// the same key already exist in this logger, it will be overridden (only for this message).
-    func warn(_ message: String, error: Error? = nil, attributes: [AttributeKey: AttributeValue]? = nil) {
-        log(level: .warn, message: message, error: error, attributes: attributes)
+        /// The service name  (default value is set to application bundle identifier)
+        public var serviceName: String?
+
+        /// The logger custom name (default value is set to main bundle identifier)
+        public var loggerName: String?
+
+        /// Enriches logs with network connection info.
+        /// This means: reachability status, connection type, mobile carrier name and many more will be added to each log.
+        /// For full list of network info attributes see `NetworkConnectionInfo` and `CarrierInfo`.
+        ///
+        /// `false` by default.
+        public var sendNetworkInfo: Bool
+
+        /// Enables the logs integration with RUM.
+        /// If enabled all the logs will be enriched with the current RUM View information and
+        /// it will be possible to see all the logs sent during a specific View lifespan in the RUM Explorer.
+        ///
+        /// `true` by default.
+        public var bundleWithRUM: Bool
+
+        /// Enables the logs integration with active span API from Tracing.
+        /// If enabled all the logs will be bundled with the `DatadogTracer.shared().activeSpan` trace and
+        /// it will be possible to see all the logs sent during that specific trace.
+        ///
+        /// `true` by default.
+        public var bundleWithTrace: Bool
+
+        /// Enables logs to be sent to Datadog servers.
+        /// Can be used to disable sending logs in development.
+        /// See also: `printLogsToConsole(_:)`.
+        ///
+        /// `true` by default.
+        public var sendLogsToDatadog: Bool
```

Comment:
blocker/ [Rule 7](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3027638506/Public+API+Convention+in+iOS+SDK+2.x#Rule-7%3A-Use-of-standard-Datadog-and-RUM-terminology): _“avoid platform self referencing”_ "Datadog" is obvious implication in this context.

It makes me think if we need this option at all. Android [`Logger` defines `sampleRate`](https://github.com/DataDog/dd-sdk-android/blob/a982b35ceaec738f19f137b6f1333af325b285a0/dd-sdk-android/src/main/kotlin/com/datadog/android/log/Logger.kt#L320-L326) and this is something we should add now or later for the sake of consistency. With `sampleRate`, distinction between "remote", "console" and "remote & console" will be clear (IMO sampling should only apply to remote logs, not console logs):
```swift
// remote:
loggerConfig.sampleRate = 100
loggerConfig.consoleLogFormat = nil

// console:
loggerConfig.sampleRate = 0
loggerConfig.consoleLogFormat = .short

// remote & console:
loggerConfig.sampleRate = 100
loggerConfig.consoleLogFormat = .short
```

If we keep `sendLogsToDatadog` option, we're allowing inconsistency:
```swift
loggerConfig.sendLogsToDatadog = false
loggerConfig.sampleRate = 100
```

If we decide to keep this option anyway, I'd suggest renaming it to:
* `remoteLogging: Bool`

WDYT @maxep @0xnm? Thoughts @xgouchet ?

Reply:
That's a good point 👍
On iOS, [sampling rate is set by the feature config](https://github.com/DataDog/dd-sdk-ios/blob/feature/v2/DatadogLogs/Sources/Logs.swift#L29) but it would be trivial to make it per logger instance.

Reply:
Indeed `sampleRate` and `sendLogsToDatadog` are linked together, but having only `sampleRate` may be not that obvious to customers (in the end, while it may sound strange, sample rate can be applied to console logs as well). Also in Android we have a `setLogcatLogsEnabled` for console logs, and `setDatadogLogsMinPriority` to control min priority of logs sent to Datadog.

So to have a symmetry with `setLogcatLogsEnabled` on Android I would keep `sendLogsToDatadog` option, but not against maybe to rename it to `remoteLogging` (then also similar term should be used for `setDatadogLogsMinPriority` method).

Reply:
Following our discussion,  I'm going for:
```swift
/// Sets the sample rate for remote logging.
///
/// **When set to `0`, no log entries will be sent to Datadog servers.**
/// A value of`100` means all logs will be processed.
///
/// When setting the `remoteSampleRate` to `0`
///
/// Default is `100`, meaning that all logs will be sent.
public var remoteSampleRate: Float

/// Set the minimum log level reported to Datadog servers.
/// Any log with a level equal or above the threshold will be sent.
///
/// Note: this setting doesn't impact logs printed to the console if `printLogsToConsole(_:)`
/// is used - all logs will be printed, no matter of their level.
///
/// `LogLevel.debug` by default
public var remoteLogThreshold: LogLevel
```